File: solution/861. Score After Flipping Matrix/861.cpp
class Solution {
public:
    int matrixScore(vector<vector<int>>& grid) {
        const int m = grid.size(), n = grid[0].size();
        for (int i = 0; i < m; ++i) {
            if (grid[i][0] == 0) {
                for (int j = 0; j < n; ++j)
                    grid[i][j] = 1 - grid[i][j];
            }
        }
        for (int j = 0; j < n; ++j) {
            int cnt = 0;
            for (int i = 0; i < m; ++i) {
                cnt += grid[i][j];
            }
            if (cnt < (m + 1) / 2) {
                for (int i = 0; i < m; ++i)
                    grid[i][j] = 1 - grid[i][j];
            }
        }
        
        int res = 0;
        for (auto& row : grid) {
            int num = 0;
            for (int j = 0; j < n; ++j)
                num = (num << 1) | row[j];
            res += num;
        }
        return res;
    }
};

File: solution/861. Score After Flipping Matrix/861.py
class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        res = (1 << n - 1) * m
        for j in range(1, n):
            cnt = sum(grid[i][j] == grid[i][0] for i in range(m))
            res += max(cnt, m - cnt) * (1 << n - j - 1)
        return res
    
